.changeset/forty-crabs-judge.md

```diff
@@ -0,0 +1,30 @@
+---
+'astro': patch
+---
+
+Adds an optional middleware for usage with `astro:env`
+
+If you're using `astro:env`, you can now use a middleware to detect server envrionment variables leaks on the client:
+
+```ts
+import { leakDetectionMiddleware } from 'astro/env/middleware'
+
+export const onRequest = leakDetectionMiddleware()
+```
+
+An error will be thrown instead of rendering the page if a leak is detected.
+
+You can pass 2 options:
+
+1. `filterContentType`: filters what response content type should trigger the check. Defaults to the content type starting with `/text` or `application/json` 
```

Comment:
Do you mean `text/*`? I am not aware of content type headers that are named `/text`